### ISSUE TITLE
[Feature Request] Quick look feature like GNOME-Sushi, or GNOME-Sushi integration

### ISSUE THREAD

**Issue by Booligoosh** [pop-os/cosmic-files#74]
Hey, thanks for the awesome work on Files. I currently use GNOME files with gnome-sushi, which does not work with COSMIC files. Gnome-sushi lets you press the space bar to preview images, PDFs, text files etc.

I was wondering what the plans are for this functionality in COSMIC files, out of the following?
1. Implementing the feature as something built into COSMIC Files (this would be awesome!!)
2. Leaving it to the community to build a cosmic-sushi app/extension
4. Implementing support for GNOME Sushi via dbus, see their [GitLab readme](https://gitlab.gnome.org/GNOME/sushi/-/blob/master/README.md?ref_type=heads):
> Sushi is a DBus-activated service. For Sushi to popup, applications will have to call the ShowFile method on the org.gnome.NautilusPreviewer2 interface.
> If you want to debug Sushi without a calling application, you can use the sushi <filename> command.

Cheers!

**Comment by Booligoosh** [comment]
Here is a DBus command that works to call GNOME Sushi:
```bash
dbus-send --session \
          --dest=org.gnome.NautilusPreviewer \
          --type=method_call \
          --print-reply \
          /org/gnome/NautilusPreviewer \
          org.gnome.NautilusPreviewer2.ShowFile \
          string:"file:///home/ethan/davmail.log" \
          string:"gnome_files" \
          boolean:true
```

If it is wanted, happy to submit a PR adding this DBus call when pressing the space bar on a file.

**Comment by maria-komarova** [comment]
#109 might potentially help with this.

**Comment by WatchMkr** [comment]
As Maria mentioned, this functionality is planned with file previews.

**Comment by adannenberg** [comment]
Reopening this issue to ask if it's still the case that "this functionality is planned with file previews".  Currently file preview shows a thumbnail rather than a full-size, scalable preview like gnome sushi.  Gnome sushi is an **enormously** useful addition designed to mimic the behavior of the spacebar-toggled preview utility in MacOS.  It would be great if the equivalent functionality were native in Cosmic or its addition didn't require a hack.

**Comment by mmstick** [comment]
@adannenberg This is already implemented, as the issue is closed. There is a keyboard shortcut (`space`) to open the image preview, or you can open the context drawer (`ctrl + space`) and click on the image preview button beside the back and forward buttons.

![Image](https://github.com/user-attachments/assets/639c984e-c2c3-4d1a-b1a3-046d92356e42)

It will then appear as so

![Image](https://github.com/user-attachments/assets/0f8361da-3797-4c8c-8bfa-9fd3322cb0c2)

**Comment by adannenberg** [comment]
This doesn't work for me and I am, as far as I know, fully updated (no
updates available in the Cosmic Store).  More specifically, when a file is
selected in the Cosmic Files app then (ctrl + space) opens the context
drawer which includes a thumbnail image and an Open button.  But (space)
has no effect...

Are you sure that the feature is implemented, i.e. in your Cosmic instance
does (space) do in the Cosmic Files app what gnome sushi does in the Ubuntu
or non-Cosmic PopOS?  If so, is there some diagnostic info that I can
provide that would help debug the issue that I have?

**Comment by l-const** [comment]
yeah, it works @adannenberg 
see: 

https://github.com/user-attachments/assets/d64fcb7d-4ce7-46ad-a12d-74bf6235deeb

**Comment by adannenberg** [comment]
On closer inspection, for me it works on png and jpg files but not on pdf,
docx or csv files.  For me, having it work for pdf files is a big deal...
 I haven't tried other file types.

**Comment by l-const** [comment]
you are right, gnome-sushi does work with .pdf files whilst cosmic-files "quick preview" doesn't

**Comment by adannenberg** [comment]
​So the issue originally opened by @Booligoosh shouldn't have been closed.  I don't permission to reopen the issue.  Can someone else reading this thread reopen the issue, or should a new issue be opened referencing this one (which seems ugly and inefficient)?

**Comment by l-const** [comment]
Yeah, all the different  viewers and the file formats that sushi supports could be seen here: https://github.com/GNOME/sushi/tree/master/src/viewers

**Comment by adannenberg** [comment]
@mmstick , can you reopen the issue?

**Comment by adannenberg** [comment]
Why does this no longer show up in either Closed or Open issues (afaict)?

**Comment by jacobgkau** [comment]
@adannenberg Not sure what you mean. It's showing up in Open issues since I re-opened it.

![Image](https://github.com/user-attachments/assets/a1756db7-6b75-46e9-af96-dd557afcf4e5)

**Comment by olegbespalov** [comment]
Hey there! I just upgraded to 24.04 and I'm truly enjoying the experience so far 🙌  

I noticed PDF previews are still missing, and I think full PDF support (like scrolling and text selection) is a bigger task, but I believe rendering a high-resolution preview of the first page would be a great first step.

I've put together a branch for this here: [olegbespalov:feat/preview-pdf](https://github.com/pop-os/cosmic-files/compare/master...olegbespalov:cosmic-files:feat/preview-pdf).

If this looks like the right direction, I'm happy to turn it into a PR. What do you think?

**Comment by maria-komarova** [comment]
@olegbespalov sorry for the delayed response, PR would make it easier for us to review the changes. Gallery previews for PDF would be nice.

**Comment by olegbespalov** [comment]
Hey @maria-komarova !

Thanks for the update!

I've opened the PR https://github.com/pop-os/cosmic-files/pull/1765

On a similar note, there is another one about text previews, and I've updated it with the latest master changes: https://github.com/pop-os/cosmic-files/pull/1675

**Comment by p-sims** [comment]
Just looking at preview on space support in round for formates beyond PDF. Current situations on Fedora Linux 43.20260903.0 is:

#### Images

- jxl: Some (#510 / #515)
- jpg: Yes
- PNG: Yes
- HIEC: No  (but preview overlay buttons appear)
- tiff: Yes
- webm: yes
- avif: No (but preview overlay buttons appear)

#### Files

- PDF: No
- .odt / .ods / .odp: No
- .log: Yes
- .json: Yes
- .md: Yes (but only as plain text, NOT in markdown formatting)
- .txt: Yes
- .rtf: No
- .sh .awk .dart .pl .php .ps1.rb: No (#1818)

#### Audio

- .opus - No
- .mp3 - No
- .m4a - No
- .wav - No
- .flac - No
- .ogg - No

#### Video (#1696)

- .mp4 - No
- .mov - No
- .mkv - No
- .webm - No
- .ogv - No
- .avi - No

#### Other

- Folders: No (should list size on disk & number of files?) (#1696)
- .tar / .gz / .zip etc - Should state size on disk?
- .ics / .ical (#1904)

Prehaps a good default, as per MacOS, is that space bar should open preview pane for all files, but where a previewer isn't avalible, it just shows filesize, full name, and maybe date created / permission...
